### ISSUE TITLE
load time for dataset created from image directory

### ISSUE THREAD

**Issue by LastHorizon** [tensorflow/datasets#3026]
/!\ IF YOU WANT PEOPLE TO HELP YOU, PLEASE GIVE AS MUCH DETAIL AS POSSIBLE, INCLUDING THE FULL STACKTRACE AND CODE SNIPPET

**Short description**
Hi all,

Firstly, apologies if this is in the wrong place but I feel this is something that would be amazing to address if possible! 
I am training a model using images stored in a directory and creating datasets with the **tf.keras.preprocessing.image_dataset_from_directory()** method. Images are set as 50 x 50 pixels, greyscale. 
My training routine is lightning fast thanks to the RTX 3090 graphics card. However, there is substantial I/O bottlenecking when filling the shuffle buffer up. Is there a way to address this and make it faster? 

Thanks! 
 
**Environment information**
* Operating System: Linux Ubuntu 18.04
* Python version: 3.6.9
* `tensorflow-datasets`/`tfds-nightly` version: 4.0.2
* `tensorflow`/`tf-nightly` version: 2.4.1

* Does the issue still exists with the last `tfds-nightly` package (`pip install --upgrade tfds-nightly`) ?

**Reproduction instructions**

```
## create data sets 

        train_ds = tf.keras.preprocessing.image_dataset_from_directory(
          data_dir,
          validation_split=validation_split,
          subset="training",
          seed=seed,
          image_size=(img_height, img_width),
          batch_size=batch_size,
          color_mode='grayscale'
        )

        val_ds = tf.keras.preprocessing.image_dataset_from_directory(
          data_dir,
          validation_split=validation_split,
          subset="validation",
          seed=seed,
          image_size=(img_height, img_width),
          batch_size=batch_size,
          color_mode='grayscale'
        )

## create and compile model 
model = get_compiled_model()

## train network (call back simply saves epoch data if epoch result is better than previous)
        history = model.fit(
          train_ds,
          validation_data=val_ds,
          epochs=epochs,
          initial_epoch=offset,
          callbacks=callbacks,
          steps_per_epoch=None,
          validation_steps=None
        )

```

**Expected behavior**
Significant decrease in time to fill shuffle buffer. 

**Additional context**

**Comment by Conchylicultor** [comment]
I would suggest to post in tensorflow/tensorflow repository. The `tf.keras` API is independent of `tensorflow_datasets` so we have no control over this. I think the shuffle buffer is only filled up during the first batch. Other batch should be faster. You can benchmark it with `tfds.benchmark(ds)`: https://www.tensorflow.org/datasets/api_docs/python/tfds/benchmark
